src/toil/batchSystems/kubernetes.py

```diff
@@ -286,6 +287,53 @@ def _try_kubernetes_expecting_gone(self, method, *args, **kwargs):
         for attempt in retry(predicate=retryable_kubernetes_errors_expecting_gone):
             with attempt:
                 return method(*args, **kwargs)
+                
+    def _try_kubernetes_stream(self, method, *args, **kwargs):
+        """
+        Kubernetes kubernetes.watch.Watch().stream() streams can fail and raise
+        errors. We don't want to have those errors fail the entire workflow, so
+        we handle them here.
+        
+        When you want to stream the results of a Kubernetes API method, call
+        this instead of stream().
+        
+        To avoid having to do our own timeout logic, we finish the watch early
+        if it produces an error.
+        """
+    
+        w = kubernetes.watch.Watch()
+        
+        # We will set this to bypass our second catch in the case of user errors.
+        userError = False
+    
+        try: 
+            for item in w.stream(method, *args, **kwargs):
+                # For everything the watch stream gives us
+                try:
+                    # Show the item to user code
+                    yield item
+                except Exception as e:
+                    # If we get an error from user code, skip our catch around
+                    # the Kubernetes generator.
+                    userError = True
+                    raise
+        except Exception as e:
```

Comment:
Hmmm... this seems to be equivalent to:

```
        except Exception as e:
            if retryable_kubernetes_errors(e):
                    # This is just cloud weather.
                    # TODO: We will also get an APIError if we just can't code good against Kubernetes. So make sure to warn.
                    logger.warning("Received error from Kubernetes watch stream: %s", e)
                    # Just end the watch.
                    return
            # Something actually weird is happening.
            raise
```

Unless `if UserError` had some special log message here?  I'm not actually completely sure what `userError` is doing here, unless I was overzealous and reviewed this PR too early.